src/Rasa.Game/Data/EntityGUID.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace Rasa.Data
+{
+    public struct EntityGUID : IEquatable<EntityGUID>
+    {
+        // Lower entity IDs are reserved in the client
+        const ulong _startingIdCounter = 1000;
+
+        public ulong Id { get; private set; }
+
+        public EntityGUID(EntityType entityType, ulong entityIdCounter)
+        {
+            Id = ((ulong)entityType & 0xFF) << 56 | ((_startingIdCounter + entityIdCounter) & 0x00FFFFFFFFFFFF);
```

Comment:
You don't have to add the starting counter to the entity id counter, counters from 1 will be unique with the proper entity type.